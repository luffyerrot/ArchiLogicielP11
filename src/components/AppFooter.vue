<template>
  <v-footer height="60" app elevation="6">
    <p class="message">Un super footer</p>
  </v-footer>
</template>

<script setup lang="ts">
</script>

<style scoped lang="scss">
    .message {
        text-align: center;
    }
</style>