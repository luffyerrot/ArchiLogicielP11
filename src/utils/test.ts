import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}