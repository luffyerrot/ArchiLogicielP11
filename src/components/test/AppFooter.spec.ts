import { factory, renderCorrectly } from "@/utils/test";
import { describe } from "vitest";
import AppFooter from "../AppFooter.vue";

describe('Footer', () => {
    const wrapper = factory(AppFooter, true);
    renderCorrectly(wrapper);
})