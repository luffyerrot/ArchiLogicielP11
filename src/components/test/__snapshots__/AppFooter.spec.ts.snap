// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > Bon affichage de la page 1`] = `"<v-footer-stub data-v-6704560a="" elevation="6" app="true" height="60" border="false" order="0" absolute="false" tile="false" tag="footer"></v-footer-stub>"`;
